tools/db_crashtest.py

```diff
@@ -663,6 +670,7 @@ def finalize_and_sanitize(src_params):
         dest_params["enable_compaction_filter"] = 0
         dest_params["sync"] = 0
         dest_params["write_fault_one_in"] = 0
+        dest_params["open_read_fault_one_in"] = 0
```

Comment:
this is the only relevant line.

Reply:
Thanks for looking into this @jaykorean ! I'm wondering if disabling the expected state based verification would be the better option? That way, we would still exercise the best-effort recovery code path that falls back to an earlier manifest on I/O error. (Note that best-effort recovery has its own dedicated set of stress test jobs, so disabling the verification wouldn't reduce the coverage for any other stress tests.)

Reply:
Thanks for looking into this @jaykorean ! I'm wondering if disabling the expected state based verification would be the better option? That way, we would still exercise the best-effort recovery code path that falls back to an earlier manifest on I/O error. (Note that best-effort recovery has its own dedicated set of stress test jobs, so disabling the verification wouldn't reduce the coverage for any other stress tests.)

Reply:
>That way, we would still exercise the best-effort recovery code path that falls back to an earlier manifest on I/O error.

Right, but if we disable expected state based verification completely when `best_efforts_recovery` = 1 and `open_read_fault_one_in` > 0, we are also skipping verifications for other operations after db reopens with recovery, no? Please correct me if I am wrong.

I was wondering if "best-effort recovery code path that falls back to an earlier manifest on I/O error" may be a better candidate for a unit test.

Reply:
>best-effort recovery has its own dedicated set of stress test jobs, so disabling the verification wouldn't reduce the coverage for any other stress tests

Ah, overlooked your comment initially. If this is the case, then disabling the verification makes more sense.

Reply:
> Right, but if we disable expected state based verification completely when `best_efforts_recovery` = 1 and `open_read_fault_one_in` > 0, we are also skipping verifications for other operations after db reopens with recovery, no? Please correct me if I am wrong.

Right, there would be no expected state based verification at all in this case. I tend to think that when it comes to BE recovery in particular, this is better than disabling read fault injection though. BE recovery was specifically built to deal with corrupt or missing database files (~I/O errors), so getting rid of I/O errors would mean we're not stressing the core logic of the feature.

> 
> I was wondering if "best-effort recovery code path that falls back to an earlier manifest on I/O error" may be a better candidate for a unit test.

Yeah, hopefully we have that already.

Reply:
>Yeah, hopefully we have that already.

I was able to confirm that we do have a unit test for checking fallback logic of manifest read failures (`DBBasicTest::BestEffortsRecoveryTryMultipleManifests`). It relies on adding a corrupted file on top of existing manifests instead of injected IO error, though. 

I want to confirm one more thing. While reading the code in `DbImpl::Recover()`, I found that we only try reading the manifest once (per file) when `best_efforts_recovery` is true. So, assuming that manifest file that we try to recover from is valid, if the read failed (but with a retryable error, such as throttled by WS), I guess we lose data that could have been recovered from the manifest. Please correct me if I am wrong here.